comms/storage/longterm/longterm.go

```diff
@@ -0,0 +1,158 @@
+// Package longterm handles taking content from the temp store and storing it in the long-term store.
+package longterm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"comms.audius.co/storage/decider"
+	"comms.audius.co/storage/transcode"
+	"github.com/nats-io/nats.go"
+	"gocloud.dev/blob"
+)
+
+type LongTerm struct {
+	storageDecider decider.StorageDecider
+	jsc            nats.JetStreamContext
+}
+
+// New creates a struct that listens to streamToStoreFrom and downloads content from the temp store to the long-term store.
+func New(streamToStoreFrom string, storageDecider decider.StorageDecider, jsc nats.JetStreamContext) *LongTerm {
+	longTerm := &LongTerm{
+		storageDecider: storageDecider,
+		jsc:            jsc,
+	}
+
+	os.MkdirAll(getFileBucketPath(), os.ModePerm)
+	longTerm.runStorer(streamToStoreFrom)
+	return longTerm
+}
+
+// Get returns a file from the long-term store.
+func (lt *LongTerm) Get(key string) (io.Reader, error) {
+	ctx := context.Background()
+	bucket, err := blob.OpenBucket(ctx, getBucketUrl())
+	if err != nil {
+		return nil, fmt.Errorf("could not open long-term bucket: %v", err)
+	}
+	defer bucket.Close()
+
+	reader, err := bucket.NewReader(ctx, key, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read %q: %v", key, err)
+	}
+	return reader, nil
+}
+
+// runStorer runs a goroutine to pull tracks from temp NATS object storage to long-term object storage.
+func (lt *LongTerm) runStorer(uploadStream string) {
+	thisNodePubKey := os.Getenv("audius_delegate_owner_wallet") // TODO: Get from config or something - same for value in NewProd() above
+	// Create a per-node explicit pull consumer on the stream that backs the track upload status KV bucket
+	storerDurable := fmt.Sprintf("STORER_%s", thisNodePubKey)
+	_, err := lt.jsc.AddConsumer(uploadStream, &nats.ConsumerConfig{
+		Durable:       storerDurable,
+		AckPolicy:     nats.AckExplicitPolicy,
+		DeliverPolicy: nats.DeliverAllPolicy, // Using the "all" policy means when a node registers it will download every track that it needs
+		ReplayPolicy:  nats.ReplayInstantPolicy,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	// Create a subscription on the consumer for every node
+	// Subject can be empty since it defaults to all subjects bound to the stream
+	storerSub, err := lt.jsc.PullSubscribe("", storerDurable, nats.BindStream(uploadStream))
+	if err != nil {
+		panic(err)
+	}
+
+	// Watch KV store to download files to long-term storage
+	// TODO: Maybe there should be an exit channel for in case we restart StorageServer without restarting the whole program? (e.g., if we want to update StorageDecider to pass it a new slice of storage node pubkeys)
+	go func() {
+		for {
+			msgs, err := storerSub.Fetch(1)
+			if err == nil {
+				msgs[0].Ack()
+
+				job := transcode.Job{}
+				err := json.Unmarshal(msgs[0].Data, &job)
+				if err != nil {
+					panic(err)
```

Comment:
log and continue for invalid input.

Reply:
good catch, thanks. any standard you prefer for logging? I was thinking of just doing `log.Printf()`, but I used `fmt.Printf()` elsewhere in this file, and in other places we use `log15`